app/uk/gov/hmrc/trackingconsentfrontend/assets/src/domain/userPreferencesFactory.ts

```diff
@@ -50,15 +60,15 @@ const userPreferencesFactory = (): UserPreferences => {
     return entries.filter(isPreferenceStrictlyBoolean)
   }
 
-  const allThePreferences = () => fromEntries(cookieTypes.map(cookieType => [cookieType, true]))
+  const allThePreferences = () => fromEntries(cookieTypes.map((cookieType) => [cookieType, true]))
 
-  const getPreferences = () => {
+  function getPreferences(): Preferences | undefined {
```

Comment:
The introduction of functions here instead of constants is a change in direction, what benefit do we get from that change?

Reply:
Ability to use `this` which is more idiomatic Javascript.

Reply:
I personally really dislike using `this` in JavaScript for stuff like (no pun intended) this.  It is more efficient but can cause problems if methods are called out of context - I find the simplicity of being able to do things like `eventEmitter.on('abc', userPreferences.userAcceptsAll)` is worth the compromise.  We can get around it with `bind`, `call` and `apply` but I don't think we gain enough to justify the inconvenience.

Reply:
Looking at this again there's actually a more subtle reason I changed it to use this. I ran it through a strict linter and saw that 'self' was being used before it was defined in line 20.

Reply:
One good option is:

```
const self = {}

self.abc = () => {}

self.def = () => {
  self.abc()
}

return self
```

Reply:
I'm not keen on mutating self if we can avoid it. What about converting to a standard ES6 class?

Reply:
Removed use of `this` in latest commit. Resolved ordering issue by declaring self at top and using functions for methods that are exported. This works because functions are hoisted. I feel this is a good compromise.